blockmanager.go

```diff
@@ -829,7 +829,13 @@ func (c *checkpointedCFHeadersQuery) requests() []*query.Request {
 // handleResponse is the internal response handler used for requests for this
 // CFHeaders query.
 func (c *checkpointedCFHeadersQuery) handleResponse(req, resp wire.Message,
-	peerAddr string) query.Progress {
+	peerAddr string, quit <-chan struct{}) query.Progress {
+
+	select {
+	case <-quit:
+		return noProgress
+	default:
+	}
```

Comment:
it doesnt need to be checked here, it should instead be checked on the `select` below on line #918. As it currently stands, things will still hang if `quit` is closed right after this `select`. 

None of the code between here and there is blocking, so can just check there which is the actual "hang" point.